model/postgres/postgres_server.rb

```diff
@@ -110,6 +110,20 @@ def configure_hash
   end
 
   def trigger_failover
+    if read_replica?
```

Comment:
Can we keep the logic more or less same for both regular cluster and read replicas somehow? For example;

```ruby
  def trigger_failover
    if representative_at && (standby = failover_target)
      standby.incr_take_over
      true
    else
      Clog.emit("Failed to trigger failover")
      false
    end
  end
```

You might need to slightly modify `taking_over` label.

Reply:
yeap!